config/routes.rb

```diff
@@ -32,10 +32,10 @@
 
   # App administration
   scope 'admin/:type', constraints: Authz::AppDeveloperConstraint do
-    resources :products, param: :name, constraints: { type: /dev|usr/ } do
+    resources :products, except: :destroy, param: :name, constraints: { type: /dev|usr/ } do
       nested do
         scope ':context' do
-          resources :permissions, only: [:index, :new, :create, :destroy], param: :name
+          resources :permissions, only: [:index, :new, :create], param: :name
```

Comment:
Why remove route for deleting permissions?

Reply:
Woops